karapace/kafka_rest_apis/authentication.py

```diff
@@ -142,6 +142,9 @@ class SimpleOauthTokenProviderAsync(AbstractTokenProviderAsync):
     async def token(self) -> str:
         return self._token
 
+    def token_with_expiry(self, _config: str | None = None) -> tuple[str, int | None]:
```

Comment:
This is necessary for now, as we still have the consumers from `aiokafka` present. Once that's replaced, we can simplify these token providers, to have a single one with this method, that doesn't really depend on the abstract classes from any library. (confluent-kafka only needs a callable as the OAuth callback)